float32 leaveDir
uint16 leaveGate